### ISSUE TITLE
Unable to generate a new PGP key from Windows app

### ISSUE THREAD

**Issue by jtsmith0101** [keybase/client#7644]
attempted to generate a new PGP key from the windows app.  Ran for about 15 minutes and I gave up.... attempted to close and the app / action isn't closing

my log id: 8a8fe1d38bd62802cf8e5e1c

**Comment by maxtaco** [comment]
Which command did you issue?

**Comment by jtsmith0101** [comment]
the "generate a new key" link / button from within the keybase windows app

**Comment by maxtaco** [comment]
I don't think that option exists from the Windows app. Can you tell me exactly what the menu/buttons said?

**Comment by jtsmith0101** [comment]
Hi Max, see attached. keybase 1 image displays my proofs, I can add a pgp key.  keybase 2 image is the next screen.  I'm selecting the top option.
![keybase 1](https://user-images.githubusercontent.com/10441596/27890769-5de775e4-6238-11e7-8333-d0b62071aa37.PNG)
![keybase 2](https://user-images.githubusercontent.com/10441596/27890770-5de8fe6e-6238-11e7-8fd3-90598e6841a2.PNG)

**Comment by maxtaco** [comment]
OK, thanks for the info. @zanderz  can you take a look?

**Comment by zanderz** [comment]
Service log shows an error generating the key:
`PGP::Generate -> ERROR: Id[0] failed to convert to PGPUserId (James T Smith (2019) <jtsmith0101@gmail.com>)`
which seemed to reach the GUI process:
```
From Keybase: log: Thu Jul 06 2017 09:23:28 GMT+1000 (AUS Eastern Standard Time) (1499297008738): PLC: [
  1499297008737,
  "Action:",
  {
    "0": "Dispatching action: profile:cancelPgpGen: ",
    "1": "{\"type\":\"profile:cancelPgpGen\"}"
  }
]
```

I tried to recreate the error by giving a different userid and email during the pgp gen step, but it sill worked for me.
Tracked internally with DESKTOP-4280

**Comment by jtsmith0101** [comment]
Nice......

I tried again, twice with success on the second try.  The first time was as
before with no luck.  The second time, however, I selected a username
without the parens "()".  This worked.  Perhaps there is a problem using
parens?

Regardless, my issue is resolved and I have a fresh certificate.

A request though, I'd like to be able to enter in an expiration date for
the certificate at creation.

Best,

James

**Comment by maxtaco** [comment]
Yeah, the front end needs to parse the `(x)` into a PGP comment.
